File: 1615-range-sum-of-sorted-subarray-sums/1615-range-sum-of-sorted-subarray-sums.py
class Solution:
    def rangeSum(self, nums: List[int], n: int, left: int, right: int) -> int:
        tmp = []
        for i in range(n):
            sum = nums[i]
            tmp.append(sum)
            for j in range(i+1, n):
                sum += nums[j]
                tmp.append(sum)
        
        tmp.sort()
        ans = 0
        mod = 10**9 + 7
        for i in range(left-1, right):
            ans = (ans + tmp[i]) % mod
        return ans

File: 1615-range-sum-of-sorted-subarray-sums/1615-range-sum-of-sorted-subarray-sums.cpp
class Solution {
public:
    int rangeSum(vector<int>& nums, int n, int left, int right) {
        vector<int> tmp;
        for(int i = 0; i < n; i++){
            int sum = nums[i];
            tmp.push_back(sum);
            for(int j = i+1; j < n; j++){
                sum+=nums[j];
                tmp.push_back(sum);
            }
        }
        sort(tmp.begin(), tmp.end());
        int ans = 0;
        const int mod = 1e9 + 7;
        for(int i = left-1; i < right; i++){
            ans = (ans + tmp[i]) % mod;
        }        
        return ans;
    }
};